used for testing

nameisbhaskar opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/125035)
<!-- Reviewable:end -->

nameisbhaskar committed:
roachtest: use spot VM for all tests

Currently, we are using spot VMs only to run benchmark tests. The reason
for the low adoption of spot VM is because of the test failures due to
VM preemption.
But, with the recent changes, we have better control on VM preemption
changes where if there is a test failure due to preemption, the test is
run on an on-demand VM the next time.
Also, the current failed tests that are run on spot VM is 148 out of 1093 = 13.5%.
If I consider failure out of total tests run is 148 out of 2508 = 5.9%.

So,  this PR removes the condition to use spot VMs only for benchmark tests.

Fixes: #127236
Epic: None

nameisbhaskar committed:
test changes
